### PR TITLE
Feature/post list options

### PR DESCRIPTION
**Pull request by scottblackburn**
## Proposed Changes
- Adds a field into the editor to change the CTA text on a post list
- Adds a toggle in the editor to hide/show excerpt for post list.

### CODE REVIEW

File: includes/blocks/list-block/render.php [hunk 1]
```diff
@@ -197,8 +197,13 @@ function benenson_list_process_content( $attributes ) {
  * @param array $data Item data.
  */
 if ( ! function_exists( 'benenson_render_list_item' ) ) {
-	function benenson_render_list_item( $data ) {
-		$title = isset( $data['title'] ) ? $data['title'] : '';
+	function benenson_render_list_item( $data, $display_excerpt, $cta_text ) {
```
Comment (jonmcp) [inline includes/blocks/list-block/render.php]: As you've changed the method signature, it'd be safest to assign default values to the new parameters (`0` and `''` respectively). Whilst this function is only called from within this file in the theme itself, it can't be guaranteed that it's not used externally by third parties; not assigning default values will cause warnings in situations where it's being called with a single parameter.

File: includes/blocks/list-block/render.php [hunk 2]
```diff
@@ -197,8 +197,13 @@ function benenson_list_process_content( $attributes ) {
...
+		$title   = isset( $data['title'] ) ? $data['title'] : '';
+		$excerpt = isset( $data['excerpt'] ) ? $data['excerpt'] : '';
+
+		if ( ! empty( $display_excerpt ) && 1 === $display_excerpt ) {
```
Comment (jonmcp) [inline includes/blocks/list-block/render.php]: If this variable is given a default value in the method signature, the `empty()` check will no longer be required.

File: includes/blocks/list-block/render.php [hunk 3]
```diff
@@ -475,9 +493,13 @@ function benenson_render_list_block( $attributes ) {
 
 			return ob_get_clean();
 		}
-
 		print '<ul class="linkList">';
-		array_map( 'benenson_render_list_item', $data );
+		$index = 0;
```
Comment (jonmcp) [inline includes/blocks/list-block/render.php]: This is unused.

File: includes/blocks/list-block/render.php [hunk 4]
```diff
@@ -475,9 +493,13 @@ function benenson_render_list_block( $attributes ) {
...
+		$total = count( $data );
```
Comment (jonmcp) [inline includes/blocks/list-block/render.php]: This is unused.

File: includes/blocks/list-block/render.php [hunk 5]
```diff
@@ -475,9 +493,13 @@ function benenson_render_list_block( $attributes ) {
...
+		foreach ( $data as $item ) {
+			benenson_render_list_item( $item, $attributes['displayExcerpt'], $attributes['ctaText'] );
+			$index++;
```
Comment (jonmcp) [inline includes/blocks/list-block/render.php]: This is unused.

File: src/scripts/blocks/post-list/DisplayComponent.js [hunk 6]
```diff
@@ -105,6 +109,21 @@ class DisplayComponent extends Component {
               checked={ attributes.categoryRelated }
               onChange={ this.createUpdateAttribute('categoryRelated') }
             />}
+            { attributes.type === 'select' && <ToggleControl
+              label={ __('Display excerpt', 'benenson') }
+              checked={ attributes.displayExcerpt }
+              onChange={ this.createUpdateAttribute('displayExcerpt') }
+            />}
```
Comment (jonmcp) [inline src/scripts/blocks/post-list/DisplayComponent.js]: Missing space before closing curly brace.

File: src/scripts/blocks/post-list/DisplayComponent.js [hunk 7]
```diff
@@ -105,6 +109,21 @@ class DisplayComponent extends Component {
...
+            { attributes.type === 'select' && (<div>
+            <label htmlFor="editorButtonText">{ __('Button Text:', 'benenson') } </label>
+             <RichText
+              name={ 'editorButtonText' }
```
Comment (jonmcp) [inline src/scripts/blocks/post-list/DisplayComponent.js]: This can be simplified to `name="editorButtonText"`.

File: src/scripts/blocks/post-list/DisplayComponent.js [hunk 8]
```diff
@@ -105,6 +109,21 @@ class DisplayComponent extends Component {
...
+              className={ ['editorButtonText'] }
```
Comment (jonmcp) [inline src/scripts/blocks/post-list/DisplayComponent.js]: This can be simplified to `className="editorButtonText"`.

File: src/scripts/blocks/post-list/DisplayComponent.js [hunk 9]
```diff
@@ -105,6 +109,21 @@ class DisplayComponent extends Component {
...
+              placeholder={ __('(CTA Text)', 'benenson') }
+              value={ attributes.ctaText }
+              keepPlaceholderOnFocus={ true }
+              onChange={ this.createUpdateAttribute('ctaText') }
+            ></RichText></div>)}
```
Comment (jonmcp) [inline src/scripts/blocks/post-list/DisplayComponent.js]: `RichText` can be self-closed: `/>` rather than `></RichText>`.
Missing space before closing curly brace.

File: src/scripts/blocks/post-list/DisplayComponent.js [hunk 10]
```diff
@@ -129,6 +148,7 @@ class DisplayComponent extends Component {
           preview={ this.state.preview }
           style={ attributes.style }
           prefix={ this.state.keyPrefix }
+          ctaText={ this.state.ctaText }
```
Comment (jonmcp) [inline src/scripts/blocks/post-list/DisplayComponent.js]: State?

File: includes/blocks/list-block/render.php [hunk 11]
```diff
@@ -475,9 +487,13 @@ function benenson_render_list_block( $attributes ) {
...
 			return ob_get_clean();
 		}
-
 		print '<ul class="linkList">';
-		array_map( 'benenson_render_list_item', $data );
+		$index = 0;
+		$total = count( $data );
+		foreach ( $data as $item ) {
+			benenson_render_list_item( $item, $attributes['displayExcerpt'] );
+			$index++;
```
Comment (ampersarnie) [inline includes/blocks/list-block/render.php]: `$index` on this line and L491 don't serve a purpose. Needs removing.

File: src/scripts/blocks/post-list/DisplayComponent.js [hunk 12]
```diff
@@ -105,6 +109,21 @@ class DisplayComponent extends Component {
...
               onChange={ this.createUpdateAttribute('categoryRelated') }
             />}
+            { attributes.type === 'select' && <ToggleControl
+              label={ __('Display excerpt', 'benenson') }
+              checked={ attributes.displayExcerpt }
+              onChange={ this.createUpdateAttribute('displayExcerpt') }
+            />}
+            { attributes.type === 'select' && (<div>
+            <label htmlFor="editorButtonText">{ __('Button Text:', 'benenson') } </label>
+             <RichText
```
Comment (jonmcp) [inline src/scripts/blocks/post-list/DisplayComponent.js]: This line appears to be mis-aligned.

File: src/scripts/blocks/post-list/DisplayComponent.js [hunk 13]
```diff
@@ -105,6 +109,21 @@ class DisplayComponent extends Component {
               checked={ attributes.categoryRelated }
               onChange={ this.createUpdateAttribute('categoryRelated') }
             />}
+            { attributes.type === 'select' && <ToggleControl
+              label={ __('Display excerpt', 'benenson') }
+              checked={ attributes.displayExcerpt }
+              onChange={ this.createUpdateAttribute('displayExcerpt') }
+            />}
+            { attributes.type === 'select' && (<div>
+            <label htmlFor="editorButtonText">{ __('Button Text:', 'benenson') } </label>
```
Comment (jonmcp) [inline src/scripts/blocks/post-list/DisplayComponent.js]: Is the trailing space after the closing curly brace intentional?

File: src/scripts/blocks/post-list/DisplayComponent.js [hunk 14]
```diff
@@ -104,7 +108,22 @@ class DisplayComponent extends Component {
...
               onChange={ this.createUpdateAttribute('categoryRelated') }
-            />}
+            /> }
+            { attributes.type === 'select' && <ToggleControl
+              label={ __('Display excerpt', 'benenson') }
+              checked={ attributes.displayExcerpt }
+              onChange={ this.createUpdateAttribute('displayExcerpt') }
+            /> }
+            { attributes.type === 'select' && (<div>
+              <label name="editorButtonText">{ __('Button Text:', 'benenson') } </label>
```
Comment (jonmcp) [inline src/scripts/blocks/post-list/DisplayComponent.js]: Revert this change; https://github.com/bigbitecreative/benenson/pull/53#discussion_r245980947 refers to L120, not L118.

**Comment by jonmcp** [comment]
You okay with this now, @ampersarnie?
